### PR TITLE
Issue #6440: AnnotationLocation: named parameters must be considered parameters

### PR DESCRIPTION
**Pull request by pbludov**
Issue #6440 

Regression report: https://pbludov.github.io/issue-6440/

### CODE REVIEW

**Comment by romani** [comment]
Strange that no changes in Google's `it`.
Please drop there some code there to have warn.

**Comment by pbludov** [comment]
> Please drop there some code there to have warn.

done

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationlocation/InputAnnotationLocationMultiple.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+import java.lang.annotation.Repeatable;
+
+/**
+ * This test-input is intended to be checked using following configuration:
+ *
+ * allowSamelineSingleParameterlessAnnotation = false
+ * allowSamelineParameterizedAnnotation = false
+ * allowSamelineMultipleAnnotations = true
+ */
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationlocation/InputAnnotationLocationMultiple.java]: Can we have some warnings in this file ?

Comment (pbludov) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationlocation/InputAnnotationLocationMultiple.java]: No. with `allowSamelineMultipleAnnotations = true` both `allowSamelineSingleParameterlessAnnotation` and `allowSamelineParameterizedAnnotation` has no effect. Actually, this means that any combination of annotations is allowed.

However, it is possible to catch a violation by using annotations on different lines.
This one: https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/annotation/AnnotationLocationCheck.java#L292
But this is a different case and should be covered by another set of tests.

**Pull request merged**